packages/wallet/src/services/ProviderStatusTracker.ts

```diff
@@ -0,0 +1,78 @@
+import { Milliseconds } from './types';
+import {
+  Observable,
+  combineLatest,
+  concat,
+  distinctUntilChanged,
+  filter,
+  map,
+  merge,
+  of,
+  share,
+  skipWhile,
+  switchMap,
+  timer
+} from 'rxjs';
+import { SyncStatus } from '../types';
+import { TrackedWalletProvider, WalletProviderFnStats } from './TrackedWalletProvider';
+import { TrackerSubject } from './util';
+
+const getDefaultWalletProviderSyncRelevantStats = (
+  walletProvider: TrackedWalletProvider
+): Observable<WalletProviderFnStats[]> =>
+  combineLatest([
+    walletProvider.stats.ledgerTip$,
+    walletProvider.stats.currentWalletProtocolParameters$,
+    walletProvider.stats.genesisParameters$,
+    walletProvider.stats.networkInfo$,
+    walletProvider.stats.queryBlocksByHashes$,
+    walletProvider.stats.queryTransactionsByAddresses$,
+    walletProvider.stats.rewardsHistory$,
+    walletProvider.stats.utxoDelegationAndRewards$
+  ]);
+
+export interface ProviderStatusTrackerProps {
+  consideredOutOfSyncAfter: Milliseconds;
+}
+
+export interface ProviderStatusTrackerDependencies {
+  walletProvider: TrackedWalletProvider;
+}
+
+export interface ProviderStatusTrackerInternals {
+  /**
+   * @returns Observable of WalletProvider stats that are considered
+   * when determining Wallet sync status
+   */
+  getWalletProviderSyncRelevantStats?: typeof getDefaultWalletProviderSyncRelevantStats;
+}
+
+export const createProviderStatusTracker = (
+  { walletProvider }: ProviderStatusTrackerDependencies,
+  { consideredOutOfSyncAfter }: ProviderStatusTrackerProps,
+  {
+    getWalletProviderSyncRelevantStats = getDefaultWalletProviderSyncRelevantStats
+  }: ProviderStatusTrackerInternals = {}
+): TrackerSubject<SyncStatus> => {
+  const relevantStats = getWalletProviderSyncRelevantStats(walletProvider);
+  const syncStatusWithoutTimeout$ = relevantStats.pipe(
+    skipWhile((allStats) => allStats.every(({ numCalls, numResponses }) => numCalls === 0 && numResponses === 0)),
+    map((allStats) =>
+      allStats.some(({ numCalls, numResponses }) => numCalls > numResponses) ? SyncStatus.Syncing : SyncStatus.UpToDate
+    ),
+    distinctUntilChanged(),
+    share()
+  );
+  const upToDate$ = syncStatusWithoutTimeout$.pipe(filter((syncStatus) => syncStatus === SyncStatus.UpToDate));
+  // const syncing$ = syncStatusWithoutTimeout$.pipe(filter((syncStatus) => syncStatus === SyncStatus.Syncing));
```

Comment:
Please explain the purpose of this commented out line

Reply:
Nice catch, removed it locally after initially pushing the branch and forgot to re-push. The comment is gone now